### PR TITLE
add choose donwload snapshot

### PR DESCRIPTION
**Pull request by Madongming**
add choose donwload snapshot

### CODE REVIEW

File: scripts/setup_fullnode.sh [hunk 1]
```diff
@@ -419,14 +419,26 @@ function main() {
...
+    fi
 
-        wantdownload=N
-        read -p "Do you prefer to start from a snapshot, Download the db file. [Y/N] (Default: N)? " wantdownload
-        if [ "${wantdownload}X" = "YX" ] || [ "${wantdownload}X" = "yX" ];then
-            # Download db file
-            donwloadBlockDataFile
+    wantdownload=Y
+    if [ "$_PLUGINS_"X = "gateway"X ];then
+        read -p "Do you prefer to start from a snapshot, This will overwrite existing data。Download the db file. [Y/N] (Default: Y)? " wantdownload
```
Comment (dustinxie) [inline scripts/setup_fullnode.sh:426]: existing data. not 。

File: scripts/setup_fullnode.sh [hunk 2]
```diff
@@ -419,14 +419,26 @@ function main() {
...
+        if [[ "$runversion" == "v1.1"* && "$version" == "v1.2"* ]];then
+            if [ "$wantdownload"X = "N"X ] || [ "$wantdownload"X = "n"X ];then
```
Comment (dustinxie) [inline scripts/setup_fullnode.sh:428]: is it possible to combine these 2 if?
if too complicated, it's fine to leave it like this

File: scripts/setup_fullnode.sh [hunk 3]
```diff
@@ -419,14 +419,26 @@ function main() {
...
         mkdir data log etc > /dev/null 2>&1
         popd
+    fi
 
-        wantdownload=N
-        read -p "Do you prefer to start from a snapshot, Download the db file. [Y/N] (Default: N)? " wantdownload
-        if [ "${wantdownload}X" = "YX" ] || [ "${wantdownload}X" = "yX" ];then
-            # Download db file
-            donwloadBlockDataFile
+    wantdownload=Y
```
Comment (dustinxie) [inline scripts/setup_fullnode.sh:424]: i think still use wantdownload=N?

File: scripts/setup_fullnode.sh [hunk 4]
```diff
@@ -419,14 +419,26 @@ function main() {
...
+    if [ "$_PLUGINS_"X = "gateway"X ];then
```
Comment (dustinxie) [inline scripts/setup_fullnode.sh:425]: and here set 
wantdownload=Y?

File: scripts/setup_fullnode.sh (hunk 3 above)
Comment (dustinxie) [reply scripts/setup_fullnode.sh:424]: so for not gateway case, the default is no need to download

**Review by dustinxie — APPROVED** [review]

**Review by Aivolt1 — COMMENTED** [review]
Good work

**Review by Aivolt1 — COMMENTED** [review]
Feedback

**Pull request merged**